src/test/java/org/opensearch/knn/index/SegmentSizeFloorMergePolicyIT.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index;
+
+import lombok.SneakyThrows;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.knn.KNNRestTestCase;
+import org.opensearch.knn.index.engine.KNNEngine;
+
+import static org.opensearch.knn.common.KNNConstants.*;
+
+public class SegmentSizeFloorMergePolicyIT extends KNNRestTestCase {
+    @SneakyThrows
+    public void testKNNIndexFloorSegmentSize() {
+        // Create a KNN index
+        String knnIndexName = "test-knn-floor-segment";
+        String mapping = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject(FIELD_NAME)
+            .field("type", "knn_vector")
+            .field("dimension", 3)
+            .startObject(KNN_METHOD)
+            .field(NAME, METHOD_HNSW)
+            .field(METHOD_PARAMETER_SPACE_TYPE, SpaceType.L2)
+            .field(KNN_ENGINE, KNNEngine.FAISS.getName())
```

Comment:
Should be Lucene engine in the test case (although this test is just changing the value, so engine being faiss won't make a difference). I'll change in next revision.

Reply:
Why does engine matter?

Reply:
It doesn't, you're right. I had the incorrect understanding that this merge policy is specific to Lucene engine, but it will also change the faiss merging behavior. I've left in the faiss engine setting.